contributors/design-proposals/federated-placement-policy.md

```diff
@@ -0,0 +1,271 @@
+    },
+    "spec": {
+      "replicas": 4,
+      "selector": {...},
+      "template": {...},
+    }
+  }
+}
+```
+
+#### Example Policy Engine Response:
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+```
+
+```json
+{
+  "result": {
+    "federation.kubernetes.io/replica-set-preferences": {
+      "clusters": {
+        "gce-europe-west1": {
+          "weight": 1
+        },
+        "gce-europe-west2": {
+          "weight": 1
+        }
+      },
+      "rebalance": true
+    }
+  }
+}
+```
+
+> This example shows the policy engine returning the replica-set-preferences.
+> The policy engine could similarly return a desired value for other annotations
+> such as the Cluster Selector annotation.
+
+#### Error Handling
+
+The admission controller is designed to **fail closed**. Request handling may
+fail because of:
+
+- Serialization errors
+- Request timeouts or other network errors
+- Unexpected errors from the policy engine
+
+In the event of request timeouts (or other network errors) or back-pressure
+hints from the policy engine, the admission controller should retry after
+applying a backoff.
+
+If the administrator has not defined a policy yet, the response from the policy
+engine indicates no annotations are defined. The admission controller does not
+treat this as an error, i.e., it admits the request.
+
+#### Configuration
+
+The admission controller requires configuration for the OPA endpoint:
+
+```
+{
+  “opa”: {
+    “baseURL”: “http://opa.federation.svc.cluster.local:8181/v1”,
+    “annotationsPath”: “/data/io/k8s/federation/annotations”,
+  }
+}
+```
+
+- `baseURL` specifies the URL to reach the policy engine.
+- `annotationsPath` specifies the path of the document that defines desired
+  values for annotations.
+
+The configuration file is provided to the federation-apiserver with the
+`--admission-control-config-file` command line argument.
+
+The admission controller is enabled in the federation-apiserver by providing the
+`--admission-control` command line argument. E.g.,
+`--admission-control=AlwaysAdmit,OPA`.
+
+### 2. Remediation of Resource Placement
+
+When policy changes or the environment in which resources are deployed changes
+(e.g. a cluster’s PCI compliance rating gets up/down-graded), resources might
+need to be moved for them to obey the placement policy. Sometimes administrators
+may decide to remediate manually, other times they may want Kubernetes to
+remediate automatically.
+
+To automatically reschedule resources onto desired clusters, we introduce a
+remediator component (**opa-kube-sync**) that is deployed as a sidecar with OPA.
+
+![Remediation](https://docs.google.com/drawings/d/1ehuzwUXSpkOXzOUGyBW0_7jS8pKB4yRk_0YRb1X4zsY/pub?w=812&h=288)
+
+The notifications sent to the remediator by OPA specify the new value for
+annotations such as replica-set-preferences.
+
+When the remediator component (in the sidecar) receives the notification it
```

Comment:
Currently the container gets a kubeconfig mounted into the container. I think this would be the same mechanism used by the federation-controller-manager?